cli/internal/package_manager/api/types.go

```diff
@@ -0,0 +1,19 @@
+package api
```

Comment:
So, I tried to do exactly this in my first pass, but ended up with Go complaining about a circular dependency because of the pattern in use here:
https://github.com/vercel/turborepo/blob/ad822724df9a4e21116a6b298543ba1674ee7dbb/cli/internal/backends/backends.go#L15-L20

The existing thing also has a split into API here:
https://github.com/vercel/turborepo/blob/ad822724df9a4e21116a6b298543ba1674ee7dbb/cli/internal/api/types.go

I'm not comfortable enough with Go code to unwind that pattern, do you have hints on what I should look up?

Reply:
I made a quick [PR](https://github.com/nathanhammond/turborepo/pull/1) to demonstrate what I meant. Feel free to adapt or use as you wish.

Reply:
Adopted your PR, but did it in the CLI. Not sure what it will eventually show in the interface.